draft-ietf-quic-transport.md

```diff
@@ -557,12 +562,20 @@ The layout of a Version Negotiation packet is:
 ~~~
 {: #version-negotiation-format title="Version Negotiation Packet"}
 
-The value in the Unused field is selected randomly by the server. The Connection
-ID field echoes the corresponding value from the triggering client packet.  This
-allows clients some assurance that the server received the packet and that the
-Version Negotiation packet is in fact from the server.  The Version field MUST
-be set to 0x00000000.  The remainder of the Version Negotiation packet is a list
-of 32-bit versions which the server supports.
+The value in the Unused field is selected randomly by the server.
+
+The server MUST include the value from the Source Connection ID field of the
+packet it receives in the Destination Connection ID field; similarly, the value
+of the Source Connection ID field is set to the Destination Connection ID from
+the received packet.  The value for Source Connection ID is randomly selected by
+a client and so gives clients some assurance that the server received the packet
+and that the Version Negotiation packet was not generated by an off-path
+attacker.
+
```

Comment:
The the client sends a random source ID, it cannot be used to route the vneg packet back to the client. It could be partially a random. Even if encrypted, it wouldn't really be random. It seems that the need for nonces somewhat conflicts with the idea of using the ID for routing. It could also make the connection ID longer than necessary just to ensure initial entropy that isn't needed subsequently.

vneg routing issue: https://github.com/quicwg/base-drafts/issues/1146

Reply:
This also conflicts with a paragraph or so lower, which says that the Source CID is "of [the client's] choosing" and later says "this might be a zero length value."

Reply:
An understandable misinterpretation.

Client sets destination to random, source to whatever it needs.
In VN, server sets destination to client's source and source to client's destination.

Thus, it's very hard to explain.  And clearly I failed.